Remove timescale from FITS format time strings

timj opened the pull request:
As discussed in #7781 and #5329.

It might be better to retain the parsing support for timescale but no longer include the timescale when converting to a string.

saimn commented:
This concerns mostly astropy.time so I will leave it to @mhvk :wink:

Cadair commented:
:+1: To keeping the parsing working with a deprecation warning.

timj commented:
Ok. I'll have another stab over the weekend.

astrofrog commented:
@timj - thanks! Just so you know, the feature freeze for v3.1 is at the end of tomorrow, so if there's any way at all you could wrap this up by then, we could be sure to include it in v3.1. There may be a small chance of getting it in over the weekend depending on when exactly the branching of v3.1.x happens though, in case you can't finish it by tomorrow evening.

Cadair commented:
Personally, I would classify this as "breaking bug fix" :stuck_out_tongue:

timj commented:
I've updated the PR so that it now understands the old style (and will parse it) but issues a deprecation warning (I added a test for it). Newly created strings will not have the time scale.

Cadair commented:
:+1:

timj commented:
I've added some extra tests to make sure things are really working and it seems fine. I've mentioned that the code section is deprecated.  I could remove the _fits_realization and _fits_scale attributes since they are now ignored, but since this code will all be deleted in the future it didn't seem worth doing.  I'll have a quick go at removing those attributes.

I agree that there is a problem now that there is no way to use the FITS deprecated timescales.  The problem seems to be that:

```
t = Time("1990-01-10T00:00:00.123", format="fits", scale="et")
```
doesn't work because the scale is handled above the formatter and is rejected before the FITS class has a chance to do anything. Fixing this would require that the Time constructor asks the formatter for a timescale translation. Not something for this PR I feel.

astrofrog commented:
@timj - thanks for enhancing the tests - can you rebase this PR to get rid of the merge conflict in the changelog?

timj commented:
Rebased and I've also removed the fits realization code and added tests that things still work the same.

bsipocz commented:
Thank you @timj!

timj commented:
I think you can close  #7781 and #5329 now.

astrofrog commented:
Thanks @timj!
